### PR TITLE
Add GitHub#{compare, sha} and use in bin/dotd.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Further PRs will further automate the `bin/dotd` script, including adding integration with the `Slack#developers` topic and adding additional `GitHub` automation.

### CODE REVIEW

File: bin/dotd [hunk 1]
```diff
@@ -110,6 +113,19 @@ def open_pull_request(base, compare, title)
...
+# Prints the commit messages of commits present in branch head but not branch
+# base.
+# @param base [String] The base branch to compare.
+# @param head [String] The head branch to compare against.
+# @example For a DTT: view_commits(base: 'test', head: 'staging')
+def view_commits(base:, head:)
+  GitHub.compare(base: base, head: head).each do |commit_message|
+    # Empirically, commit messages often contain blank lines. For console
+    # readability, we attempt to remove them.
+    puts commit_message.gsub("\n\n", "\n")
```
Comment (islemaster) [inline bin/dotd:125]: I wonder if it'd be nicer to just print the first line of each commit.  Commit messages conventionally have a 'topic' line, a blank line, and more content to be friendly to summary output like this:
```
* dc98101 - (HEAD -> minitest-junit-reporting, origin/minitest-junit-reporting) Add JUnit reports for cucumber [test all] (16 hours ago) <Brad Buchanan>
* 029f1da - Move apps JUnit report to apps/ directory (16 hours ago) <Brad Buchanan>
* 3ca2e8b - Add JUnit reports for shared tests on Circle CI [test all] (19 hours ago) <Brad Buchanan>
* cf4b04f - Provide JUnit reports for dashboard tests on Circle [test all] (19 hours ago) <Brad Buchanan>
*   5d84067 - Merge pull request #13630 from code-dot-org/section-projects-sorting (2 days ago) <David Bailey>
```
I know a lot of my commits are one-line messages but for a handful I practically write the whole PR description - I wouldn't want someone to have to deal with that here.

File: lib/cdo/github.rb [hunk 2]
```diff
@@ -70,4 +74,28 @@ def self.html_url(pr_number)
...
+  # @raise [Exception] From calling Octokit.compare.
+  # @example For a DTT, compare(base: 'test', head: 'staging').
+  # @return [Array[String]] The commit messages of all commits between base and
+  #   head.
+  def self.compare(base:, head:)
+    base_sha = sha(base)
+    head_sha = sha(head)
+
+    response = Octokit.compare(REPO, base_sha, head_sha)
+    response.commits.map(&:commit).map(&:message)
```
Comment (islemaster) [inline lib/cdo/github.rb:90]: :heart_eyes: _That is so cool_

**Comment by ashercodeorg** [comment]
Per feedback in `Slack#developers`, I'm going to add another commit to this PR that branches behavior based on whether `CDO.github_access_key` is available.

**Comment by ashercodeorg** [comment]
Actually, after further conversation, it seems reasonable to require DOTDs to have an access token. Merging.

File: bin/dotd (hunk 1 above)
Comment (ashercodeorg) [reply bin/dotd:125]: It would be nicer, I'll implement this suggestion in a follow up PR.

**Pull request merged**